.github/workflows/changed_file_trigger_examples_check.yml

```diff
@@ -0,0 +1,109 @@
+      - name: setup matrix
+        id: set-matrix
+        run: |
+          changedFileName=""
+          for file in ${{ steps.changed-files.outputs.all_changed_files  }}; do
+            changedFileName="${file}:${changedFileName}"
+          done
+          echo "$changedFileName was changed"
+          res=`python .github/workflows/changed_file.py --fileNameList $changedFileName`
+          echo "All changed files are $res"
+          loc=$( IFS=',' ; echo "${res[*]}" )
+          echo "$loc"
+          echo "::set-output name=matrix::{\"loc\":$(echo "$loc")}"
+
+# If no file is changed, it will prompt an error and shows the matrix do not have value.
+  check-all-changed-files:
+    if: ${{ github.event_name == 'pull_request' }}  # Add this to avoid when schedule triggers the workflow, this should keep silent.
+    name: Test each changed example files
+    needs: detect-changed-files
+    runs-on: [self-hosted, gpu]
+    strategy:
+      matrix: ${{fromJson(needs.detect-changed-files.outputs.matrix)}}
+    container:
+      image: hpcaitech/pytorch-cuda:1.12.0-11.3.0
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+      - name: Install dependancies
+        run: |
+          pip install -r ./requirements/requirements.txt
+          pip install colossalai
+      - name: List all changed example files
+        run: |
+          res=${{ matrix.loc }}
+          cd "${PWD}/examples/${res}"
+          bash test_ci.sh
+          cd ../../..
+#          pip install -e .
+
+# This is for all files' weekly check
+  matrix_preparation:
+    if: ${{ github.event_name == 'schedule' }}
+    name: Prepare Directory List for All files
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+    - name: 📚 Checkout
+      uses: actions/checkout@v3
+    - name: setup matrix
+      id: set-matrix
+      run: |
+        res=`python .github/workflows/weekly_check.py`
+        all_loc=$( IFS=',' ; echo "${res[*]}" )
+        echo "$all_loc"
+        echo "::set-output name=matrix::{\"all_loc\":$(echo "$all_loc")}"
+
+  weekly_check:
+    if: ${{ github.event_name == 'schedule' }}
+    name: Weekly check all examples
+    needs: matrix_preparation
+    runs-on: [self-hosted, gpu]
+    strategy:
+      matrix: ${{fromJson(needs.matrix_preparation.outputs.matrix)}}
+    container:
+      image: hpcaitech/pytorch-cuda:1.12.0-11.3.0
+    steps:
+      - name: 📚 Checkout
+        uses: actions/checkout@v3
+      - name: Install the requirements
+        run: |
+          pip install -r ./requirements/requirements.txt
+          pip install colossalai
+      - name: Traverse all files
+        run: |
+          dir=${{ matrix.all_loc }}
+          echo "${dir} is current directory"
+          cd "${PWD}/examples/${dir}"
+          bash test_ci.sh
+          cd ../../..
+#          pip install -e .
```

Comment:
Lines 108 and 109 are not needed any more.

Reply:
OK, done!